app/flatpak-builtins-build.c

```diff
@@ -434,27 +410,27 @@ flatpak_builtin_build (int argc, char **argv, GCancellable *cancellable, GError
         }
 
       *split++ = 0;
-      add_args (argv_array,
-                "--bind", split, opt_bind_mounts[i],
-                NULL);
+      flatpak_bwrap_add_args (bwrap,
+                              "--bind", split, opt_bind_mounts[i],
+                              NULL);
     }
 
   if (opt_build_dir != NULL)
     {
-      add_args (argv_array,
-                "--chdir", opt_build_dir,
-                NULL);
+      flatpak_bwrap_add_args (bwrap,
+                              "--chdir", opt_build_dir,
+                              NULL);
     }
 
-  g_ptr_array_add (argv_array, g_strdup (command));
+  g_ptr_array_add (bwrap->argv, g_strdup (command));
   for (i = 2; i < rest_argc; i++)
-    g_ptr_array_add (argv_array, g_strdup (argv[rest_argv_start + i]));
+    g_ptr_array_add (bwrap->argv, g_strdup (argv[rest_argv_start + i]));
```

Comment:
(continued)
Some index magic remains here, but if that could be avoided maybe similar to flatpak_bwrap_append_args() used instead?  (Maybe the type of argv also does not match - I'm no glib expert...)

Reply:
Yeah, lots of further cleanup is possible to. In fact, i would like the entire "actually exec bwrap" code to be moved into FlatpakBwrap too. Baby steps though.